DataFormats/CTPPSReco/interface/TotemRPLocalTrack.h

```diff
@@ -0,0 +1,169 @@
+    
+        FittedRecHit() : TotemRPRecHit(), residual_(0), pull_(0) {}
+    
+        virtual ~FittedRecHit() {}
+    
+        inline const TVector3 & getGlobalCoordinates() const { return space_point_on_det_; }
+        inline void setGlobalCoordinates(const TVector3 & space_point_on_det) { space_point_on_det_ = space_point_on_det; }
+    
+        inline double getResidual() const { return residual_; }
+        inline void setResidual(double residual) { residual_ = residual; }
+    
+        inline double getPull() const { return pull_; }
+        inline void setPull(double pull) { pull_ = pull; }
+    
+        inline double getPullNormalization() const { return residual_ / pull_; }
+    
+      private:
+        TVector3 space_point_on_det_; ///< mm
+        double residual_;             ///< mm
+        double pull_;                 ///< normalised residual
+    };
+
+  public:
+    ///< parameter vector size
+    static const int dimension = 4;
+
+    ///< covariance matrix size
+    static const int covarianceSize = dimension * dimension;
+
+    TotemRPLocalTrack() : z0_(0), chiSquared_(0), valid_(false)
+    {
+    }
+
+    TotemRPLocalTrack(double z0, const TVectorD &track_params_vector,
+      const TMatrixD &par_covariance_matrix, double chiSquared);
+
+    virtual ~TotemRPLocalTrack() {}
+
+    inline const edm::DetSetVector<FittedRecHit>& getHits() const { return track_hits_vector_; }
+    inline void addHit(unsigned int detId, const FittedRecHit &hit)
+    {
+      track_hits_vector_.find_or_insert(detId).push_back(hit);
+    }
+
+    inline double getX0() const { return track_params_vector_[0]; }
+    inline double getX0Sigma() const { return sqrt(CovarianceMatrixElement(0, 0)); }
+    inline double getX0Variance() const { return CovarianceMatrixElement(0, 0); }
+
+    inline double getY0() const { return track_params_vector_[1]; }
+    inline double getY0Sigma() const { return sqrt(CovarianceMatrixElement(1, 1)); }
+    inline double getY0Variance() const { return CovarianceMatrixElement(1, 1); }
+
+    inline double getZ0() const { return z0_; }
+    inline void setZ0(double z0) { z0_ = z0; }
+
+    inline double getTx() const { return track_params_vector_[2]; }
+    inline double getTxSigma() const { return sqrt(CovarianceMatrixElement(2, 2)); }
+
+    inline double getTy() const { return track_params_vector_[3]; }
+    inline double getTySigma() const { return sqrt(CovarianceMatrixElement(3, 3)); }
+
+    inline TVector3 getDirectionVector() const
+    {
+      TVector3 vect(getTx(), getTy(), 1);
+      vect.SetMag(1.0);
+      return vect;
+    }
+
+    TVectorD getParameterVector() const;
+    void setParameterVector(const TVectorD & track_params_vector);
+
+    TMatrixD getCovarianceMatrix() const;
+    void setCovarianceMatrix(const TMatrixD &par_covariance_matrix);
+
+    inline double getChiSquared() const { return chiSquared_; }
```

Comment:
please check DataFormats/TrackReco/interface/TrackBase.h
at least as an inspiration for method names.
If we can't reuse the base, let's at least try to keep more similar interface.